cmake/HPX_AdaptWindowsPaths.cmake

```
@@ -0,0 +1,12 @@
+# Copyright (c) 2020 ETH Zurich
+#
+# SPDX-License-Identifier: BSL-1.0
+# Distributed under the Boost Software License, Version 1.0. (See accompanying
+# file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+function(adapt_path_if_win var)
+  if(WIN32)
+    string(REPLACE "\\" "/" new_var "${${var}}")
```

I think this needs to be crafted a bit more carefully. We shouldn't replace `\\` (i.e. an already escaped backslash with '//'). The following might be a starting point (from the top of my head, not verified):
```suggestion
    # replace any backslash/forward-slash that is not followed by another
    # backslash/forward-slash
    string(REGEX REPLACE "[\\\\/][^\\\\/]*$" "/" newvar "${${var}}")
```
Optionally you could now replace all remaining (now escaped) backslashes with a single forward-slash as well.

---

When I did that I based it on the fact that do a `cd <name1>//<name2>` is equivalent to do a `cd <name1>/<name2>` on Linux, but I actually haven't check on windows..
I think I'll go with your suggestion of the `TO_CMAKE_PATH` and then replace semi column thanks!